src/main/java/gov/usgs/util/CryptoUtils.java

```diff
@@ -195,37 +175,138 @@ public static Cipher getDecryptCipher(final Key key)
 	}
 
 	/**
-	 * A convenience method that chooses a signature algorithm based on the key
-	 * type. Works with DSA and RSA keys.
+	 * Create and configure a signature object based on key type.
+	 *
+	 * @param key
+	 *     Key used to sign/verify.
+	 * @param version
+	 *     SIGNATURE_V1 or SIGNATURE_V2
+	 * @return
+	 *     Configured Signature object
+	 * @throws InvalidKeyException
+	 * @throws NoSuchAlgorithmException
+	 * @throws SignatureException
 	 */
-	public static String sign(final PrivateKey privateKey, final byte[] data)
+	public static Signature getSignature(final Key key, final Version version)
 			throws InvalidKeyException, NoSuchAlgorithmException,
-			SignatureException {
-		if (privateKey instanceof DSAPrivateKey) {
-			return sign(DSA_SIGNATURE_ALGORITHM, privateKey, data);
-		} else if (privateKey instanceof RSAPrivateKey) {
-			return sign(RSA_SIGNATURE_ALGORITHM, privateKey, data);
+					SignatureException {
+		Signature signature = null;
+		if (version == Version.SIGNATURE_V1) {
+			if (key instanceof DSAPrivateKey || key instanceof DSAPublicKey) {
+				signature = Signature.getInstance(DSA_SIGNATURE_ALGORITHM);
+			} else if (key instanceof RSAPrivateKey || key instanceof RSAPublicKey) {
+				signature = Signature.getInstance(RSA_SIGNATURE_ALGORITHM);
+			}
+		} else if (version == Version.SIGNATURE_V2) {
+			if (key instanceof DSAPrivateKey || key instanceof DSAPublicKey) {
+				signature = Signature.getInstance(SIGNATURE_V2_DSA_ALGORITHM);
+			} else if (key instanceof RSAPrivateKey || key instanceof RSAPublicKey) {
+				signature = Signature.getInstance(SIGNATURE_V2_RSA_ALGORITHM);
+			}
 		} else {
-			throw new InvalidKeyException("Expected a DSA or RSA key.");
+			throw new IllegalArgumentException("Unexpected signature version " + version);
+		}
+		return signature;
```

Comment:
Should we verify the signature is not null prior to returning?

Reply:
Added check and throws